data/transactions/logic/eval.go

```diff
@@ -2654,51 +2668,47 @@ func opGaidImpl(cx *EvalContext, giw uint64, opName string) (sv stackValue, err
 	if aid := cx.TxnGroup[gi].ApplyData.ApplicationID; aid != 0 {
 		return stackValue{Uint: uint64(aid)}, nil
 	}
-	err = fmt.Errorf("%s: index %d did not create anything", opName, gi)
-	return
+	return sv, fmt.Errorf("%s: index %d did not create anything", opName, gi)
 }
 
-func opGaid(cx *EvalContext) {
+func opGaid(cx *EvalContext) error {
 	gi := uint64(cx.program[cx.pc+1])
 	sv, err := opGaidImpl(cx, gi, "gaid")
 	if err != nil {
-		cx.err = err
-		return
+		return err
 	}
 
 	cx.stack = append(cx.stack, sv)
+	return nil
 }
 
-func opGaids(cx *EvalContext) {
+func opGaids(cx *EvalContext) error {
 	last := len(cx.stack) - 1
 	gi := cx.stack[last].Uint
 
 	sv, err := opGaidImpl(cx, gi, "gaids")
 	if err != nil {
-		cx.err = err
-		return
+		return err
 	}
 
 	cx.stack[last] = sv
+	return nil
 }
 
 func (cx *EvalContext) getRound() (rnd uint64, err error) {
```

Comment:
Agree on removing them, and wonder if there is agreement in what I've found - using the named return values in combination with naked "return" is way harder to understand at a glance. It also somehow tempts the author (as here) to "flow through" more .  Just assign to the `err` and continue.  Precious braincells are them wasted trying to make sure that's correct.  I've moved to returning explicitly almost always.

Reply:
That makes a lot of sense. With named return values, I believe there's also a risk of variable shadowing if you do `err := ...` inside of a block in the function. Go's compiler seems to be good at detecting when this is problematic, but not in all cases, so it takes me a little more mental effort to review cases where I worry this might be happening.